pkg/storage/ingest/reader.go

```diff
@@ -974,64 +953,66 @@ func (r *concurrentFetchers) getEndOffset(ctx context.Context) (int64, error) {
 }
 
 func (r *concurrentFetchers) runFetchers(ctx context.Context, startOffset int64) {
-	defer level.Info(r.logger).Log("msg", "done running fetchers")
-	wg := sync.WaitGroup{}
-	wg.Add(r.concurrency)
-	defer wg.Wait()
+	fetchersWg := &sync.WaitGroup{}
```

Comment:
[nit] Have you considered `dskit_concurrency.ForEachJob` here?

Reply:
i tried it, but it's not working great - i still need a wait group because ForEachJob blocks and needs to be run in a separate goroutine at which point it's the same as running our own goroutines